accepted/exception-handling.md

```diff
@@ -0,0 +1,540 @@
+     `exn_args...`.
+
+  The semantics are as follows:
+
+  * The `func` function is invoked, passing `args...` as
+    arguments.
+  * When the invocation returns normally, control is transferred to
+    the block named by `ok_label`, passing first the call's return
+    values and then the `ok_args...`  values.
+  * When the invocation is unwound due to an exception, there are two
+    sub-cases to consider:
+    1. If the runtime determines that the exception matches one of the
+       tags for an entry in the `ir::ExceptionTable`, then control is
+       transferred to the entry's associated `exn_label` block,
+       passing the exception payload and any associated
+       `exn_args...`.
+    2. Otherwise, unwinding continues past this function's frame and
+       control is not transferred to either the `ok_label` or the
+       `exn_label`.
+
+2. A corresponding indirect-call variant:
+
+   ```
+   try_call_indirect sig, v123(args...), ok_label(ok_args...), [ tag => exn_label(exn_args...), ... ]
+   ```
+
+   This variant has the same validity requirements and semantics as
+   `try_call` but instead of taking a static `ir::FuncRef`, it takes a
+   function signature immediate and dynamic function address operand
+   value.
+
+These new instructions have many immediates and operands. Likely more
+than will naively fit in an `ir::InstructionData`. We leave the exact
+details of bitpacking and encoding these instructions to the
+implementation.
+
+Note that we do not define a CLIF instruction for throwing an
+exception. Instead, exception throwing must be done indirectly via an
+imported function (e.g. a Wasmtime builtin libcall implemented in the
+host/engine).
+
+Furthermore, we do not define `try_return_call[_indirect]`
+instructions. When you tail call out of a function, that function is
+no longer on the stack to catch any exceptions unwinding the stack.
+
+Signatures do not need to be annotated with whether they can raise
+exceptions and, if so, which exceptions they raise. A regular
+`call[_indirect]` that unwinds due to an exception will simply
+continue unwinding over its frame. One only needs to use `try_call`
+(and pay for the additional register spills) when one is catching an
+exception, not for every call that might raise an exception.
+
+We also do not define a `throw` or `raise` instruction. It is assumed
+that all exceptions will be thrown by calling out to runtime routines
+that invoke the unwinder directly. Thus, all CLIF instructions where
+an exception could be propagated are variants of `call`
+instructions. That, in turn, means that our `try_call[_indirect]`
+instructions are sufficient to enumerate all exceptional control
+edges.
+
+We do not define support for two-phase exceptions (as [used by
+C++](https://nicolasbrailo.github.io/blog/2013/0326_Cexceptionsunderthehood8twophasehandling.html))
+where landing pads can be resumed multiple times. It should be
```

Comment:
Two phase exceptions doesn't run langing pads multiple times. Only the personality function.